File: 4_b.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
int fib(int n)
{
    if (n <= 1)
    {
        return n;
    }
    return fib(n - 1) + fib(n - 2);
}
int main()
{
    int terms;
    printf("Enter the number of terms:");
    scanf("%d", &terms);
    printf("Fibonacci Series: ");
    clock_t start = clock();
    for (int i = 0; i < terms; i++)
    {
        printf("%d, ", fib(i));
    }
    clock_t end = clock();

    double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;
    printf("\nTime taken to fibonacci series (Recursive) %d elements: %f seconds: ", terms, time_taken);

    printf("\n");
    return 0;
}

File: 4_b.py
import matplotlib.pyplot as plt

# Data
n_values = [10,20,30,40,50]
time_values = [0.000000, 0.016000, 0.032000, 1.077000, 122.574000, 
               ]

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(n_values, time_values, marker='o', linestyle='-', color='blue')

# Titles and labels
plt.title('Fibonacci Series(Recursive): nth no vs Time Taken')
plt.xlabel('nth no (n)')
plt.ylabel('Time Taken (seconds)')
plt.grid(True)

# Show data point values
for i in range(len(n_values)):
    plt.text(n_values[i], time_values[i] + 0.00001, f"{time_values[i]:.6f}", 
             ha='center', fontsize=8)

plt.tight_layout()
plt.show()